network/network_integration_test.go

```diff
@@ -846,13 +847,17 @@ func startNode(t *testing.T, name string, testDirectory string, opts ...func(cfg
 	}
 
 	eventPublisher := events.NewManager()
-	if err := eventPublisher.(core.Configurable).Configure(core.ServerConfig{Datadir: testDirectory}); err != nil {
+	serverConfig := core.ServerConfig{Datadir: path.Join(testDirectory, name)}
+	if err := eventPublisher.(core.Configurable).Configure(serverConfig); err != nil {
 		t.Fatal(err)
 	}
 	if err := eventPublisher.(core.Runnable).Start(); err != nil {
 		t.Fatal(err)
 	}
 
+	storeProvider := storage.New()
```

Comment:
it might be wise to invest in a `storage.NewTestProvider(t *testing.T)`

Reply:
There actually is (`NewTestStorageEngine()`), I'll apply it